MineSweeper.py

```diff
@@ -0,0 +1,261 @@
+import Tkinter as tk
+import random
+
+import sys
+sys.path.append('~/Library/Python/2.7/site-packages')
+
+from drawingpanel import *
+
+BOARD_HEIGHT = 300
+BOARD_WIDTH = 300
+BOX_SIZE = 20
+
+NUMBER_MINES = ((BOARD_WIDTH / BOX_SIZE - 1)**2)//10 
+
+class Game():
+
+  def __init__(self):
+    self._master = tk.Tk()
+    self._panel = Canvas(self._master, width = BOARD_WIDTH, height = BOARD_HEIGHT + 100, bg = 'white')
+    self._panel.pack()
+
+    self._board = Board(self._panel)
+
+    self._board.assign_mines()
+
+    self._master.bind("<Button-1>", self.left_click_event)
+    self._master.bind("<Button-2>", self.right_click_event)
+
+     
+  def game_over(self):
+
+    self._master.unbind("<Button-1>")
+    self._master.unbind("<Button-2>")
+
+    self._b = Button(self._master, text = 'Play Again?', command = self.new_game)
+    self._panel.create_text(BOARD_WIDTH/2, BOARD_HEIGHT + 40, text  = 'Game Over', fill = 'red') 
+    self._b.pack()
+
+  def new_game(self):
+
+    self.__init__()
+
+    self._board.draw()
+
+    self._master.mainloop()
+  # reveals the tile that was clicked
+
+  def left_click_event(self, event):
+ 
+    mouse_click = []
+
+    mouse_click.append(event.x)
+    mouse_click.append(event.y)
+
+    box_index_x = (mouse_click[0] - 10) // BOX_SIZE
+    box_index_y = (mouse_click[1] - 10) // BOX_SIZE
+
+    try:
+      self._board._tiles[box_index_x][box_index_y].show_self()
+
+      if(self._board._tiles[box_index_x][box_index_y].value == 0):
```

Comment:
This is an example that isn't okay. This one is reaching into the private details of `_board` to get information. You can actually tell this by the way you named your variables. `_` is a pattern for languages that don't have the concept of private and public information. If you're calling a method on something that isn't `self` with a leading `_`, you should be wary.

Here, you could say "well... Game needs to know if a tile's value is a bomb... but it doesn't have access directly. What does have access? The board!" So you could write `Board#location_is_bomb(x, y) => bool`, and here, instead of this if, you could write `if (self._board.location_is_bom(box_index_x, box_index_y)`.